### ISSUE TITLE
Constant calico node health endpoint failure logs

### ISSUE THREAD

**Issue by c-knowles** [kubernetes-incubator/kube-aws#1381]
Using 0.9.10 with self hosted canal, it seems the canal nodes log this almost continuously:

```
2018-06-28 17:04:35.753 [ERROR][27] health.go 192: Health endpoint failed, trying to restart it... error=listen tcp :9099: bind: address already in use
```

Looks similar to https://github.com/kubernetes/kubernetes/issues/56670 and https://github.com/kubernetes/kubernetes/issues/49263#issuecomment-384754398 and https://github.com/kubernetes/kubernetes/issues/61901. k8s 1.9.8.

**Comment by c-knowles** [comment]
@davidmccormick are you seeing anything similar on your clusters with self hosted canal?

**Comment by davidmccormick** [comment]
Hmm, haven’t had any problems with the networking but will check the logs on Monday to be sure.  We’re running 1.9, haven’t upgraded to 1.10 except for a couple of test clusters.

**Comment by c-knowles** [comment]
@davidmccormick great, thanks. Actually the cluster appears to be functioning still, I don't know the what the impact of these error logs is but surely it must be something.

**Comment by c-knowles** [comment]
I was looking through this today and found the changes from https://github.com/kubernetes-incubator/kube-aws/pull/1323 in latest master. Specifically the nodeAffinity part which looked like it might help since I think before that change canal node and canal master were trying to take the same host ports. I've just updated a dev cluster with that but still using k8s 1.9 and the error logs have disappeared.

Going to close since this is already fixed in master. Note that the latest released binary of v0.10.0 suffers from this problem when canal self hosting is enabled. @mumoshu I've applied milestone v0.11.0 assuming this will be contained there.
